### DISCUSSION TITLE
Is it possible to change components style from its Traits?

### DISCUSSION THREAD

**Discussion by mohamad97-alhomsi** [GrapesJS/grapesjs#4751] [Q&A]
I'm using grapes js JSON  object
so in JSON object, each component has a type, attributes, and components fields in its object 
we know attributes field has the component traits values so I use these traits from the attributes field 
What do I need to make these traits affect components in the editor like if i was using style manager on this component 
**Note**: i didn't use style manager to change components' style because i cannot access its values like padding or color, .....etc 
from attributes fields 
**Exp:** : if i create trait for a component named rounding corners so if i change values for this trait i want to change the component style and make the component has rounded corners as my trait value
I hope someone understands what I mean

**Comment by ronaldohoch** [comment]
Hello!
```js

  editor.TraitManager.addType('traitname', {
    createInput({ trait }) {
      // Your UI
    },
    onEvent({ elInput, component, event }) {
       // some code
       component.addAttributes({ css })
    }
  }
```

I think you can use `addAttributes` to the component or use the `getSelectedToStyle()` inside onEvent from trait

```js
editor.getSelectedToStyle().setStyle({
  ...editor.getSelectedToStyle().attributes.style,
  color:"red"
})
```
